analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -343,6 +343,22 @@ public void InfiniteForLoop(string arg)
             }
         }
 
+        public void LoopWithIfsInside(bool condition)
```

Comment:
I would like to see some positive impact of this change to a rule. Otherwise, there's not really a reason to do it.

What problem does it actually solve?

Reply:
The impact is minimal. If you review by commit, you can see that one edge-case FP was fixed. The main reason is to simplify the code in `Binary.cs`.